### PR TITLE
fix(rpc): "bytes" memcmp encoding in getProgramAccounts and programSubscribe

### PR DESCRIPTION
**Pull request by prestonsn**
Add missing Agave-compatible memcmp filter support for `getProgramAccounts` and `programSubscribe`, including raw byte arrays and `encoding: "bytes"` behavior.

Also covers Agave size limits and invalid encoding/value edge cases with shared parser tests.

### Testing 

| Area | Case | Result |
|---|---|---|
| HTTP `getProgramAccounts` | base58 default string | Sig matched Agave |
| HTTP `getProgramAccounts` | base64 string | Sig matched Agave |
| HTTP `getProgramAccounts` | raw array default | Sig matched Agave |
| HTTP `getProgramAccounts` | raw array with `encoding: "bytes"` | Sig matched Agave |
| HTTP `getProgramAccounts` | raw array with `encoding: "base64"` | Sig matched Agave |
| HTTP `getProgramAccounts` | string with `encoding: "bytes"` | Sig matched Agave |
| HTTP `getProgramAccounts` | invalid encodings and oversized/invalid bytes | Sig matched Agave accept/reject |
| HTTP `getProgramAccounts` | real SPL Token programdata memcmp | Both returned SPL Token program |
| WS `programSubscribe` | valid raw-array memcmp | Both accepted |
| WS `programSubscribe` | invalid raw value `[256]` | Both rejected |

### CODE REVIEW

File: src/rpc/jrpc_websockets/methods.zig [hunk 1]
```diff
@@ -15,7 +15,7 @@ pub const DataSlice = sig.rpc.account_codec.DataSlice;
 pub const TransactionDetails = sig.rpc.methods.common.TransactionDetails;
 
 pub const MAX_PROGRAM_FILTERS: usize = 4;
-const MAX_MEMCMP_BYTES = 128;
+const MAX_MEMCMP_BYTES = sig.rpc.filters.MAX_DATA_SIZE;
```
Comment (prestonsn) [inline src/rpc/jrpc_websockets/methods.zig]: Consolidate and import from rpc.filters since they are defined there.

File: src/rpc/jrpc_websockets/methods.zig [hunk 2]
```diff
@@ -184,23 +184,15 @@ pub const ProgramSubscribe = struct {
...
         ) std.json.ParseFromValueError!Memcmp {
-            const parsed = try std.json.innerParseFromValue(Parsed, allocator, source, options);
-            const enc = parsed.encoding orelse .base58;
+            const parsed = try sig.rpc.filters.Memcmp.jsonParseFromValue(allocator, source, options);
             return .{
                 .offset = parsed.offset,
-                .bytes = try decodeFilterBytes(allocator, parsed.bytes, enc),
+                .bytes = parsed.bytes,
             };
         }
```
Comment (prestonsn) [inline src/rpc/jrpc_websockets/methods.zig]: Now that we're just delegating to `sig.rpc.filters.Memcmp.jsonParseFromValye`, can this `jsonPraseFromValue` be removed?

File: src/rpc/jrpc_websockets/methods.zig (hunk 1 above)
Comment (prestonsn) [reply src/rpc/jrpc_websockets/methods.zig]: https://github.com/Syndica/sig/pull/1458/commits/04586db2d7a088800564e91dc6400188787917cb

File: src/rpc/jrpc_websockets/methods.zig (hunk 2 above)
Comment (prestonsn) [reply src/rpc/jrpc_websockets/methods.zig]: https://github.com/Syndica/sig/pull/1458/commits/04586db2d7a088800564e91dc6400188787917cb

**Review by ajw221 — APPROVED** [review]
LGTM

**Review by jbuckmccready — APPROVED** [review]
